### PR TITLE
breaks messages of more than 50 blocks into multiple messages

### PR DESCRIPTION
**Pull request by colestrode**
Slack messages can only contain up to 50 blocks (https://api.slack.com/reference/block-kit/blocks). This PR will take all fully constructed blocks required for the message, then split them into groups of at most 50 and post them one at a time in sequence. This allows us to monitor even more repos and get more alerts! Just what everyone wants!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 node_modules
-*.bak
\ No newline at end of file
+*.bak
+.env
```
Comment (colestrode) [inline .gitignore:3]: added this so we can have local env vars stored

File: index.js [hunk 2]
```diff
@@ -11,93 +11,128 @@ const githubClient = new GithubClient(token);
...
+      },
+    ],
+  });
 
-    if (process.env.POST_TO_SLACK === 'true') {
-        await slackClient.postMessage({ blocks });
-    } else {
-        console.log(`Slack blocks: ${JSON.stringify(blocks,null,2)}`)
+  if (process.env.POST_TO_SLACK === "true") {
+    const allBlocks = breakBlocks(blocks);
```
Comment (colestrode) [inline index.js:52]: this through line 56 are the primary change. `breakBlocks` returns an array of arrays of blocks. We post those one at a time in sequence

File: index.js [hunk 3]
```diff
@@ -11,93 +11,128 @@ const githubClient = new GithubClient(token);
...
+    ],
+  };
+}
+
+/**
+ * Slack only allows 50 blocks per message, break array of all blocks into groups of at most 50.
+ * Returns an array of arrays of blocks in order.
+ * @param {*} blocks
+ */
+function breakBlocks(blocks) {
```
Comment (colestrode) [inline index.js:120]: this function is the other change I made to break the whole list of blocks into smaller arrays of 50 or less. It returns an array of arrays of blocks.

File: index.js [hunk 4]
```diff
@@ -11,93 +11,131 @@ const githubClient = new GithubClient(token);
...
+  });
 
-    if (process.env.POST_TO_SLACK === 'true') {
-        await slackClient.postMessage({ blocks });
-    } else {
-        console.log(`Slack blocks: ${JSON.stringify(blocks,null,2)}`)
+  if (process.env.POST_TO_SLACK === "true") {
+    const allBlocks = breakBlocks(blocks);
+    // await will work with oldschool loops, but nothing requires a callback like array.forEach()
+    for (let i = 0; i < allBlocks.length - 1; i++) {
```
Comment (nancyhabecker) [inline index.js:54]: `<` - should be `<=`

**Comment by nancyhabecker** [comment]
tested locally - just had the one comment re: the for loop but otherwise looks good!

File: index.js (hunk 4 above)
Comment (colestrode) [reply index.js:54]: lol < and -1... I'm rusty!

**Pull request merged**